chore: add documentation for useTimeZoneConversion [LIBS-396]

tomzemp opened the pull request:
Adds documentation for the `useTimeZoneConversion` hook.

I have tried to cover most applicable use cases without too much additional explanation.

Some other points (like javascript dates sucking and this hopefully becoming irrelevant when temporal is released) might be handled in a blog post @Topener? FYI: The original ticket ([LIBS-396](https://dhis2.atlassian.net/browse/LIBS-396)) and the [PR for the code](https://github.com/dhis2/app-runtime/pull/1308) also has a lot of background discussion. 

[LIBS-396]: https://dhis2.atlassian.net/browse/LIBS-396?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Topener commented on docs/hooks/useTimeZoneConversion.md:
````diff
@@ -0,0 +1,117 @@
+```jsx
+import React from 'react'
+import moment from 'moment'
+import { useDataQuery } from '@dhis2/app-runtime'
````
looks like `useTimeZoneConversion` needs to be added here too

Topener commented on docs/hooks/useTimeZoneConversion.md:
```diff
@@ -0,0 +1,117 @@
+| **getServerZonedISOString** | _method (returns string)_ | method that takes no arguments and returns a string representation of the wall-clock time according to the server in ISO format. For example, if it is 3 February 2023 at 12:00 in the server time zone, `myDate.getServerZonedISOString()` will return `2023-02-03T12:00:00.000`                                                                   ...
+| **getClientZonedISOString** | _method (returns string)_ | method that takes no arguments and returns a string representation of the wall-clock time according to the browser in ISO format. For example, if it is 3 February 2023 at 12:00 in the server time zone (Africa/Freetown), but we are using our DHIS2 instance in Oslo, there will be a 1-hour time difference, and `myDate.getClientZonedISOStrin ...
+
+**NOTE**: Both `fromServerDate` and `fromClientDate` will return a DHIS2Date object corresponding to the _client/browser_ time zone. This is done so that the time encapsulated by the DHIS2Date object is the true time; JavaScript Date objects are ultimately the number of milliseconds since 1 January 1970 UTC. As such, the representation will be browser specific based on the browser's time zone. Th ...
```
This is a very long-winded note that is hard to follow; perhaps it needs a little restructuring/rewriting?

Topener commented on docs/hooks/useTimeZoneConversion.md:
```diff
@@ -0,0 +1,117 @@
+
+### fromClientDate / simulating server wall-clock time
+
+As mentioned above, `fromClientDate` will return a DHIS2Date object that corresponds to the client/browser's time zone. The primary purpose of the `fromClientDate` function is to allow one to get a string representation representing the server time zone (as illustrated in the previous example). However, in certain advanced cases, you may need to initialize a date that behaves such that its wall-c ...
```
Same as above, very long, hard to follow. It might need a little restructuring.

Topener commented on docs/hooks/useTimeZoneConversion.md:
```diff
@@ -0,0 +1,117 @@
+        <div>
+            <h3>Indicators (first 10)</h3>
+            {error && <span>{`ERROR: ${error.message}`}</span>}
+            {data && lastLogin && (
```
This probably needs to be `lastLoginClient`

tomzemp commented:
Thanks for the review @Topener. I have fixed the code examples (and added a note to the last one to make it clearer that it is purely an example and shouldn't be copy/pasted).

Would you like to try doing the rewording for the points that you thought could benefit? I think the time zone issues are a bit complicated to communicate in a clear manner, so maybe it would benefit from your attempts to reformulate? 

Thanks

Topener commented:
@tomzemp yes I'll get on that! Thanks for the improvements

Topener commented:
@tomzemp I've improved the text some, please check if I didn't break anything flow wise, otherwise we're good

tomzemp commented:
Great. Thanks @Topener 🙏. Could you run prettier on that file as it's now failing the lint check?
